doc/dsl/interpretation_laufzeit.md

```diff
@@ -1,13 +1,323 @@
+
+![UML: Erzeugung Komponentenprototyp](img/create_component_prototype.png){width="50%"}
+
+**Evaluierung von Ausdrücken**
+
+Um Prototypen zu erzeugen, müssen die rechtsseitigen Ausdrücke einer Eigenschaftszuweisung (z.B.
+`x_velocity: 2.0`) evaluiert werden.
+Im obigen Beispiel handelt es sich bei diesen Ausdrücken um triviale Dezimalzahlen, es könnte sich allerdings bei
+rechtsseitigen Ausdrücken auch um Funktionsaufrufe, Verweise auf globale Objekte, etc. handeln (vgl. für
+gültige Ausdrücke hierzu [Ausdrücke](sprachkonzepte.md#ausdrücke)).
+
+Der `DSLInterpreter` ist ein [Tree-Walk-Interpreter](https://craftinginterpreters.com/a-tree-walk-interpreter.html),
+läuft also über den AST und führt für jeden so besuchten Knoten Operationen aus. Für alle Knoten, die Teil eines
+Ausdrucks sein können, erzeugt der Interpreter eine `Value`-Instanz, die den Wert des besuchten Knotens enthält.
+
+Für `NumNode` wird bspw. einfach der Wert des AST-Knotens in ein `Value` verpackt:
+```java
+public Object visit(NumNode node) {
+    return new Value(BuiltInType.intType, node.getValue());
+}
+```
+
+Für einen `GameObjectDefinition`-Knoten, der Teil eines Ausdrucks ist, ist dieses Vorgehen deutlich komplexer und wird
+unter [Typinstanziierung](#typinstanziierung) genauer erläutert. Allerdings wird auch für diesen Fall ein `Value`-Objekt
+zurückgegeben.
+
+**Anmerkung:**
+
+Die im Folgenden beschriebenen Aspekte bzgl. `quest_config` als zentralem Übergabepunkt von DSL -> Dungeon sind WIP
+und können sich daher noch grundlegend ändern (siehe hierzu [Issue #195](https://github.com/Programmiermethoden/Dungeon/issues/195)).
+
+Der zentrale Übergabepunkt zwischen dem DSL Programm und dem Dungeon-Framework wird durch die
+`quest_config`-Definition gebildet. Daher sucht der `DSLInterpreter` im nächsten Schritt die erste `quest_config`-Definition
+aus einem DSL Programm heraus und evaluiert alle Eigenschaftszuweisungen. Hierdurch werden nur Objekte und Definitionen
+evaluiert, die in Eigenschaftszuweisungen dieser `quest_config`-Definition referenziert werden.
+
+Abschließend erzeugt der `DSLInterpreter` eine `QuestConfig`-Instanz und gibt diese an das Dungeon-Framework zurück.
+Die `QuestConfig`-Instanz enthält alle Informationen für das Dungeon-Framework, um ein Dungeonlevel mit spezifizierten
+Entitäten (als `game_object`-Definition) zu erzeugen.
+
+**Wann und wie interpretiert der Interpreter?**
+
+Die Evaluierung der `quest_config`-Definition ist die **einzige** Art der Interpretation, die der `DSLInterperter`
+standardmäßig durchführt.
+Es ist aktuell kein **eigenständiger** DSL-Loop oder ähnliches vorgesehen, der parallel bzw. mit dem Gameloop des Dungeons
+mitläuft und kontinuierlich Teile des DSL-Programms ausführt.
+Die weitere Tätigkeit des `DSLInterpreter`s beschränkt sich auf die Interpretation der Event-Handler DSL-Funktionen, die
+mit Entitäten verknüpft wurden (siehe dazu [Funktionsaufrufe](#funktionsaufrufe)).
```

Comment:
Das wirkt auf mich wie eine Art Einleitung? Sollte das weiter nach oben in der Struktur?

Reply:
Ist weiter nach [oben](https://github.com/Programmiermethoden/Dungeon/pull/328/files#diff-c385e188d19dc0b1c9cea0bbf43ab942f6ced43a3fceaaa2b96caf4d8d097c5eR21-R30) gerutscht.